src/integration_test/java/com/klodnicki/Bike/v2/rest/controller/AdminChargingStationControllerIntegrationTest.java

```diff
@@ -0,0 +1,137 @@
+    @Autowired
+    private BikeRepository bikeRepository;
+    @Autowired
+    private ModelMapper modelMapper;
+    private ChargingStation chargingStation;
+    private Bike bike;
+
+    @BeforeEach
+    public void setUp() {
+        chargingStation = new ChargingStation(null, "station name", "station address", "station city",
+                100, new ArrayList<>());
+        chargingStationRepository.save(chargingStation);
+
+        bike = new Bike(null, BikeType.ELECTRIC, null, null, null);
+        bikeRepository.save(bike);
+    }
+
+    @Test
+    public void addBikeToList_ShouldAddBikeToStationListOfBikesAndReturnChargingStation_WhenStationIdAndBikeIdAreGiven() {
+        List<Bike> bikeList = new ArrayList<>();
+        bikeList.add(bike);
+        chargingStation.setBikeList(bikeList);
+
+        webTestClient.post()
+                .uri(uriBuilder -> uriBuilder
+                        .path("/api/admin/stations/list/add/" + chargingStation.getId())
+                        .queryParam("bikeId", bike.getId())
+                        .build())
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(ChargingStation.class)
+                .consumeWith(response -> {
+                            ChargingStation station = response.getResponseBody();
+                            assertNotNull(station);
+                            assertFalse(station.getBikeList().isEmpty());
+                            assertIterableEquals(chargingStation.getBikeList(), station.getBikeList());
+                        }
+                );
+    }
+
+    @Test
+    public void add_ShouldAddChargingStationToDatabaseAndReturnStationForAdminResponseDTO_WhenChargingStationIsGiven() {
+        StationForAdminResponseDTO expected = modelMapper.map(chargingStation, StationForAdminResponseDTO.class);
+
+        webTestClient.post()
+                .uri("/api/admin/stations/add")
+                .bodyValue(chargingStation)
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(StationForAdminResponseDTO.class)
+                .consumeWith(response -> {
+                            StationForAdminResponseDTO stationDTO = response.getResponseBody();
+                            assertNotNull(stationDTO);
+                            assertEquals(expected.getId(), stationDTO.getId());
+                            assertEquals(expected.getName(), stationDTO.getName());
+                            assertEquals(expected.getAddress(), stationDTO.getAddress());
+                            assertEquals(expected.getCity(), stationDTO.getCity());
+                            assertEquals(expected.getFreeSlots(), stationDTO.getFreeSlots());
+                        }
+                );
+    }
+
+    @Test
+    public void findById_ShouldReturnStationForAdminResponseDTO_WhenStationIdIsGivenAndStationExistsInDatabase() {
```

Comment:
The name of this test - part with "should" - is not correct. The method that is tested will always return this object type (unless the exception is thrown). Could you provide me with better name examples for this test?

Reply:
maybe findById_ReturnsMappedStationForAdminResponseDTO_WhenValidIdIsProvided() or findById_MapsAndReturnsStationForAdminResponseDTO_WhenStationExists()?